Mage.Sets/src/mage/cards/t/TorrentOfLava.java

```diff
@@ -0,0 +1,128 @@
+package mage.cards.t;
+
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.dynamicvalue.common.ManacostVariableValue;
+import mage.abilities.effects.common.DamageAllEffect;
+import mage.abilities.effects.common.PreventDamageToSourceEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAllEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Zone;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.Predicates;
+import mage.filter.predicate.mageobject.AbilityPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.stack.Spell;
+
+import java.util.UUID;
+
+/**
+ * @author Cguy7777
+ */
+public final class TorrentOfLava extends CardImpl {
+
+    private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("creature without flying");
+
+    static {
+        filter.add(Predicates.not(new AbilityPredicate(FlyingAbility.class)));
+    }
+
+    public TorrentOfLava(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{X}{R}{R}");
+
+        // Torrent of Lava deals X damage to each creature without flying.
+        this.getSpellAbility().addEffect(new DamageAllEffect(ManacostVariableValue.REGULAR, filter));
+
+        // As long as Torrent of Lava is on the stack, each creature has
+        // "{tap}: Prevent the next 1 damage that would be dealt to this creature by Torrent of Lava this turn."
+        this.addAbility(new SimpleStaticAbility(Zone.STACK, new TorrentOfLavaGainAbilityEffect()));
+    }
+
+    private TorrentOfLava(final TorrentOfLava card) {
+        super(card);
+    }
+
+    @Override
+    public TorrentOfLava copy() {
+        return new TorrentOfLava(this);
+    }
+}
+
+class TorrentOfLavaGainAbilityEffect extends GainAbilityAllEffect {
+
+    TorrentOfLavaGainAbilityEffect() {
```

Comment:
Save lava card id here as param

Reply:
There's probably something obvious I'm missing;  I'm not sure how I would get the correct id here.

PS: This class was based off [`GrothamaAllDevouringGainAbilityEffect()`](https://github.com/magefree/mage/blob/184cf0f4d99ddc207be0a4a2de298599053615dc/Mage.Sets/src/mage/cards/g/GrothamaAllDevouring.java#L62C75-L62C75).